apps/sms/views/conversation/js/conversation.js

```diff
@@ -1084,7 +1087,7 @@ var ConversationView = {
   close: function conv_close() {
     return this._onNavigatingBack().then(() => {
       this.cleanFields();
-      ActivityHandler.leaveActivity();
+      return (new ActivityClient()).postResult();
```

Comment:
I'm adding it in server part if result is not specified, not sure if it's the best place though :)